test/quic_multistream_test.c

```diff
@@ -4325,6 +4325,46 @@ static const struct script_op script_71[] = {
     OP_END
 };
 
+/* 72. Version negotiation: QUIC_VERSION_1 ignored */
+static int server_gen_version_neg(struct helper *h, const struct script_op *op)
+{
+    QUIC_CHANNEL *ch = ossl_quic_tserver_get_channel(h->s);
+
+    if (!TEST_true(ossl_quic_channel_test_generate_version_neg(ch,
```

Comment:
Couldn't this be done with a fault injector datagram listener, to add the version neg packet directly to the datagram? This would keep the test code out of the main library which I think we should generally be avoiding where possible.

Reply:
We could replace the initial packet from the server with the version negotiation packet. The server should cope with it shouldn't it?

Reply:
Yeah, basically replace the contents of an existing datagram with new contents...or you could even just send the datagram directly.

Reply:
> Just to note, this is currently blocked on QUIC_TLS raising the traffic secrets at the same time — the retransmission that occurs when replacing an existing datagram currently doesn't work right without it.

See #21810